src/dodal/devices/fast_grid_scan.py

```diff
@@ -91,6 +91,15 @@ def _get_y_axis(cls, y_axis: GridAxis, values: dict[str, Any]) -> GridAxis:
     def _get_z_axis(cls, z_axis: GridAxis, values: dict[str, Any]) -> GridAxis:
         return GridAxis(values["z2_start"], values["z_step_size"], values["z_steps"])
 
+    @validator("dwell_time", always=True)
+    def non_ineger_dwell_time(cls, dwell_time: float):
+        nums_a = dwell_time * 1000
+        nums_b = np.floor(dwell_time * 1000)
+        num_is_close = np.isclose(nums_a, nums_b, rtol=1e-1)
+        if not num_is_close:
+            raise ValueError("Non integer value")
```

Comment:
Should: This isn't a very descriptive exception, would be good to say what's not an integer and what value it is.